spyder/app/tests/test_mainwindow.py

```diff
@@ -1437,6 +1437,59 @@ def test_run_cython_code(main_window, qtbot):
     main_window.editor.close_file()
 
 
+@pytest.mark.slow
+def test_project_path(main_window, tmpdir, qtbot):
+    """Test project path added to spyder_pythonpath and IPython Console."""
+    projects = main_window.projects
+
+    # Create a project path
+    path = str(tmpdir.mkdir('project_path'))
+    assert path not in projects.get_conf('spyder_pythonpath', section='main')
+
+    # Ensure project path is added to spyder_pythonpath
+    projects.open_project(path=path)
+    assert path in projects.get_conf('spyder_pythonpath', section='main')
+
+    # Ensure project path is added to IPython console
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    control = shell._control
+    qtbot.waitUntil(
+        lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("import sys; sys.path")
+    assert path in control.toPlainText()
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("clear")
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("import os; os.environ.get('PYTHONPATH')")
+    assert path in control.toPlainText()
+
+    projects.close_project()
+
+    # Ensure that project path is removed from spyder_pythonpath
+    assert path not in projects.get_conf('spyder_pythonpath', section='main')
+
+    # Ensure that project path is removed from IPython console
+    shell = main_window.ipyconsole.get_current_shellwidget()
+    control = shell._control
+    qtbot.waitUntil(
+        lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("import sys; sys.path")
+    assert path not in control.toPlainText()
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("clear")
+
+    with qtbot.waitSignal(shell.executed):
+        shell.execute("import os; os.environ.get('PYTHONPATH')")
+    assert path not in control.toPlainText()
```

Comment:
Just in case, a suggestion to make the test pass (replaicing comparing to the console control widget plain text to instead check a variable retrieved from the shell):
```suggestion
    qtbot.waitUntil(
        lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)

    with qtbot.waitSignal(shell.executed):
        shell.execute("import sys; paths = sys.path")
    assert path in shell.get_value("paths")

    with qtbot.waitSignal(shell.executed):
        shell.execute("clear")

    with qtbot.waitSignal(shell.executed):
        shell.execute("import os; paths = os.environ.get('PYTHONPATH')")
    assert path in shell.get_value("paths")

    projects.close_project()

    # Ensure that project path is removed from spyder_pythonpath
    assert path not in projects.get_conf('spyder_pythonpath', section='main')

    # Ensure that project path is removed from IPython console
    shell = main_window.ipyconsole.get_current_shellwidget()
    qtbot.waitUntil(
        lambda: shell._prompt_html is not None, timeout=SHELL_TIMEOUT)

    with qtbot.waitSignal(shell.executed):
        shell.execute("import sys; paths = sys.path")
    assert path not in shell.get_value("paths")

    with qtbot.waitSignal(shell.executed):
        shell.execute("clear")

    with qtbot.waitSignal(shell.executed):
        shell.execute("import os; paths = os.environ.get('PYTHONPATH', [])")
    assert path not in shell.get_value("paths")
```

Reply:
Thanks, @dalthviz. I was not aware of `shell.get_value`. I pushed up a more compact version of your suggestion.